### PR TITLE
Add negative and positive unary operators

### PR DESCRIPTION
**Pull request by Schaeff**
The goal is to be able to write `-42` instead of `0 - 42`.

The positive operator is added for completeness.

This applies to field and uint.

Todo:
- [x] Enforce precedence
- [x] More precedence tests

### CODE REVIEW

File: zokrates_parser/src/zokrates.pest [hunk 1]
```diff
@@ -84,7 +86,8 @@ spread_or_expression = { spread | expression }
 range_or_expression = { range | expression }
 array_initializer_expression = { "[" ~ expression ~ ";" ~ constant ~ "]" }
 
-unary_expression = { op_unary ~ term }
+signeable_expression = { "(" ~ expression ~ ")" | primary_expression }
+exponent_expression = { "(" ~ expression ~ ")" | primary_expression }
```
Comment (JacobEberhardt) [inline zokrates_parser/src/zokrates.pest:95]: `signeable_expression` is redundant with 'exponent_expression' and not referred to elsewhere in the grammar. Why is it needed?

Comment (Schaeff) [reply zokrates_parser/src/zokrates.pest:88]: I thought I removed that! Good shout!

Comment (Schaeff) [reply zokrates_parser/src/zokrates.pest:95]: It is not and I actually removed it just before you posted this :p

Comment (Schaeff) [reply zokrates_parser/src/zokrates.pest:95]: I don't know why this doesn't show as outdated, maybe have a look at the latest diff here and close this if you're cool!

**Comment by Schaeff** [comment]
@dark64 this is finally ready for a review

**Review by dark64 — APPROVED** [review]
LGTM!

**Pull request merged**